### PR TITLE
perf: icon shadow optimization

### PR DESCRIPTION
**Pull request by jjspace**
ref #205

**Description**
This will greatly reduce the size of the compiled css files by not having to duplicate the `box-shadows` property for every size of each icon. This change makes it so everything about the icon is based off of the `font-size` property by using `em` units. A wonderful side effect is that now icons and pixel-art can be any size you want if you just change the font-size on the `.nes-icon` all the `.is-small` etc. classes do is adjust between predefined sizes. The width and hight is calculated according to the number of pixels in each dimension.

**Before:**

Build time: 0m31.281s

| Size | file |
| -----|-----|
| 633K | nes.css |
|  64K | nes.css.map |
| 602K | nes.min.css |

**After:**

Build Time: 0m25.455s

| Size | file |
| -----|-----|
| 255K | nes.css |
|  46K | nes.css.map |
| 237K | nes.min.css |

**Compatibility**
This shouldn't break anything, should be tested everywhere to make sure.

**Caveats**

I would like to discuss making all the pixel-arts based off the `.nes-icon` class. Because of this change most of the properties that were variable before are the same as `.nes-icon` and across the board so could probably benefit from inheriting from a root class and overriding the 2 or 3 that still need to be changed. I can provide specific examples if people want to look into this change more.

### CODE REVIEW

**Comment by jjspace** [comment]
I'd like to expand upon the caveat I listed above. After this change is pushed through the only differences that all pixel art classes have with a `nes-icon` is the `width`, `height` and `font-size` properties. In my opinion these could start inherited from the root `.nes-icon` class and overridden if the pixel-art is a different size than the standard 16x16 of icons. The diff below shows what a standard icon and pixel art class currently looks like showing they are nearly identical.
Imgur
You'll notice there is also a difference with the `display: block` property but this does not seem to effect the result so could be added or removed when combined with no penalty.

I brought this topic up in the Gitter chat and this is what @BcRikko had to say about it.

> ダーシノ @BcRikko Dec 18 22:11
> @jjspace `pixel-arts` is a little playground.🤣
> These icons are typically rarely used.
> so `pixel-arts` has no `.nes-icon` and size style. 👍 
>
> ダーシノ @BcRikko Dec 18 22:16
> `.nes-pixel-art` (or `.nes-art`) might be better, because I'd like to clearly splite `16x16 icon` and others.
> ```html
> <i class="nes-icon is-heart" />
> 
> <i class="nes-pixel-art is-octocat" />
> <i class="nes-art is-octocat" />
> ```

I think this totally fair and I agree that people will likely rarely use the pixel art classes but this PR is all about optimizing and cleaning up the icons/pixel arts and I see this as one other spot where things could be optimized. Either way works for me I just wanted to get others' input on it while we're at it. Specifically interested in what @guastallaigor might have to say as they made most of the pixel arts that we have today.

**Comment by BcRikko** [comment]
This PR is genius. 😍 

but all icons are the same size in my environment 😭

File: scss/utilities/icon-mixin.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@mixin pixelize($matrix, $colors, $size, $default-color: null) {
+@mixin pixelize($matrix, $colors, $default-color: null) {
```
Comment (guastallaigor) [inline scss/utilities/icon-mixin.scss]: I think being able to reduce the size of the entire framework is amazing :tada: ! To me it's really clever what you've done here, I never thought about it, just awesome work. 

I've created most of the pixel arts just for the fun of it and maybe to attract more users :smile: .
In my environment, I've forked your repo to test it and I notice that you changed the `icon-mixin.scss`, but checkboxes, lists and radio buttons use that aswell and they got bugged here, not sure if I'm missing something.

Radio:

image

Checkbox:

image

Lists:

image

I've tested all the rest and LGTM.

Comment (jjspace) [reply scss/utilities/icon-mixin.scss]: Should be fixed now, I forgot to set their `font-size` and remove the size from the `pixelize` call

**Comment by guastallaigor** [comment]
LGTM :+1:

**Pull request merged**